src/main/webapp/app/course/learning-paths/components/learning-path-lecture-unit/learning-path-lecture-unit.component.html

```diff
@@ -0,0 +1,25 @@
+@if (loading()) {
+    <div class="row justify-content-center m-4">
+        <div class="spinner-border text-primary" role="status">
+            <span class="sr-only">Loading...</span>
+        </div>
+    </div>
+} @else if (lectureUnit()) {
+    @switch (lectureUnit()!.type) {
+        @case (LectureUnitType.VIDEO) {
+            <jhi-video-unit [videoUnit]="lectureUnit()!" (onCompletion)="setLearningObjectCompletion($event)" />
+        }
+        @case (LectureUnitType.TEXT) {
+            <jhi-text-unit [textUnit]="lectureUnit()!" (onCompletion)="setLearningObjectCompletion($event)" />
+        }
+        @case (LectureUnitType.ATTACHMENT) {
+            <jhi-attachment-unit [attachmentUnit]="lectureUnit()!" (onCompletion)="setLearningObjectCompletion($event)" />
+        }
+        @case (LectureUnitType.EXERCISE) {
+            <jhi-exercise-unit [exerciseUnit]="lectureUnit()!" />
+        }
+        @case (LectureUnitType.ONLINE) {
+            <jhi-online-unit [onlineUnit]="lectureUnit()!" (onCompletion)="setLearningObjectCompletion($event)" />
```

Comment:
One thing I'm noticing is that we have lots of `!`s in the template code, even when it's wrapped in `@if(someSignal())`. 

This makes sense from typescript's point of view (as it wouldn't know that the value hasn't changed between calls), but could you check if there's a way around this? 🤔

Reply:
This is just due to the null safety properties in this case. As the individual unit components require an input of type LectureUnit I have to require the non null type. So unfortunately in this case I could not find another way.